db/wal_manager.cc

```diff
@@ -321,9 +321,11 @@ Status WalManager::GetSortedWalsOfType(const std::string& path,
           s = env_->GetFileSize(archived_file, &size_bytes);
           if (!s.ok() && env_->FileExists(archived_file).IsNotFound()) {
             // oops, the file just got deleted from archived dir! move on
-            s = Status::OK();
             continue;
           }
+        } else {
+          // the file just got deleted from WAL folder, skip it
```

Comment:
I would feel more comfortable if you stored the original FileExists status and asserted here that the status IsNotFound.  Otherwise, you are assuming the status is either OK or NotFound and there is not some other issue.

Reply:
sure, make sense.